views/components/table_card.erb

```diff
@@ -0,0 +1,43 @@
+<% headers = defined?(headers) && headers ? headers : [] %>
+
+<div class="overflow-hidden rounded-lg shadow ring-1 ring-black ring-opacity-5 bg-white divide-y divide-gray-200">
+  <table class="min-w-full divide-y divide-gray-300">
+    <% unless headers.empty? %>
+      <thead class="bg-gray-50">
+        <tr>
+          <% headers.each_with_index do |(name, opts), i| %>
+            <th
+              scope="col"
+              class="<% if i == 0 %> pl-4 pr-3 sm:pl-6 <% elsif i + 1 == headers.length %> pl-3 pr-4 sm:pr-6 <% else %> px-3 <% end %> py-3.5 text-left text-sm font-semibold text-gray-900 <%= opts&.dig(:classes)%>"
+            ><%= name %></th>
+          <% end %>
+        </tr>
+      </thead>
+    <% end %>
+    <tbody class="divide-y divide-gray-200 bg-white">
+      <% rows.each do |columns| %>
+        <tr>
+          <% columns.each_with_index do |(value, opts), i| %>
```

Comment:
Maybe we could do `opts ||= {}` here.  That way you could avoid the `opts &&` below.  That allocates rows*columns hashes, but you can avoid that by defining `Clover::OPTS = {}.freeze`, and doing `opts ||= OPTS`.  That's an optimization approach used by both Sequel and Roda.